lib/gui/etcher.js

```diff
@@ -124,29 +134,67 @@ const createMainWindow = () => {
   })
 }
 
-electron.app.on('window-all-closed', electron.app.quit)
-
-// Sending a `SIGINT` (e.g: Ctrl-C) to an Electron app that registers
-// a `beforeunload` window event handler results in a disconnected white
-// browser window in GNU/Linux and macOS.
-// The `before-quit` Electron event is triggered in `SIGINT`, so we can
-// make use of it to ensure the browser window is completely destroyed.
-// See https://github.com/electron/electron/issues/5273
-electron.app.on('before-quit', () => {
-  process.exit(EXIT_CODES.SUCCESS)
-})
-
 settings.load().then((localSettings) => {
   Object.assign(config, localSettings)
 }).catch((error) => {
   // TODO: What do if loading the config fails?
   console.error('Error loading settings:')
   console.error(error)
 }).finally(() => {
-  if (electron.app.isReady()) {
+  // Enable touch events if your device supports them
+  if (config.touch) {
+    app.commandLine.appendSwitch('--touch-devices')
+  }
+
+  // Simulate touch events - might be useful for touchscreen with partial driver support
+  if (config.touchSimulate) {
+    app.commandLine.appendSwitch('--simulate-touch-screen-with-mouse')
+  }
+
+  // Listen for a 'resin-update-lock' to either enable, disable or check
+  // the update lock from the renderer process (i.e. the app)
+  if (config.resinUpdateLock) {
+    const lockFile = require('lockfile')
+    ipcMain.on('resin-update-lock', (event, command) => {
+      switch (command) {
+        case 'lock':
+          lockFile.lock('/tmp/resin/resin-updates.lock', (error) => {
```

Comment:
Hmm, not `/tmp/balena/updates.lock` ? https://www.balena.io/docs/learn/deploy/release-strategy/update-locking/
Perhaps it'd be worth adding a `lockFilePath` const (instead of repeating the filename in multiple places), to make it easier to switch in future?

Reply:
@lurch Good catch! We'll also try to find a proper place for the default settings in `write-env-config.js`, this is just a stub to speed up development of etcher-pro stuff so many things will change and many will need to be adjusted